### PR TITLE
Bugfix/modaltrigger

### PR DESCRIPTION
**Pull request by EloyID**
Changes
a) The way the DatatableFilter passed the filter symbol react object didn't let add onClick, now it can
b) Material UI ModalTrigger accepts trigger as prop ass bootstrap's does

### CODE REVIEW

File: packages/vulcan-core/lib/modules/components/Datatable/DatatableFilter.jsx [hunk 1]
```diff
@@ -62,7 +62,7 @@ const DatatableFilter = props => {
           />
         }
         size="small"
-        trigger={<Filter count={getCount(columnFilters)} />}>
+        trigger={Filter({ count: getCount(columnFilters) })}>
```
Comment (eric-burel) [inline packages/vulcan-core/lib/modules/components/Datatable/DatatableFilter.jsx]: I don't get this change sorry. Why would you need to call the React component directly as a function? That could provoke performance issuses

Comment (eric-burel) [reply packages/vulcan-core/lib/modules/components/Datatable/DatatableFilter.jsx]: More precisely what's the behaviour/error you encounter when directly using a React component?

**Comment by eric-burel** [comment]
Can I close this PR and merge only #2545 ?

File: packages/vulcan-core/lib/modules/components/Datatable/DatatableFilter.jsx (hunk 1 above)
Comment (EloyID) [reply packages/vulcan-core/lib/modules/components/Datatable/DatatableFilter.jsx]: Il was the only way for the trigger to accept the onClick to open the modal.

**Comment by eric-burel** [comment]
Merged through #2545

**Pull request closed without merging**